ffi/capi/src/timezone.rs

```diff
@@ -151,25 +152,73 @@ pub mod ffi {
         #[diplomat::rust_link(icu::time::zone::ZoneNameTimestamp::far_in_past, FnInStruct, hidden)] // documented
         pub fn at_date_time_iso(&self, date: &IsoDate, time: &Time) -> Box<Self> {
             Box::new(Self {
-                zone_name_timestamp: Some(icu_time::zone::ZoneNameTimestamp::from_date_time_iso(
-                    icu_time::DateTime {
-                        date: date.0,
-                        time: time.0,
-                    },
-                )),
+                zone_name_timestamp: Some(
+                    icu_time::zone::ZoneNameTimestamp::from_zoned_date_time_iso(
+                        icu_time::ZonedDateTime {
+                            date: date.0,
+                            time: time.0,
+                            zone: self.offset.unwrap_or(icu_time::zone::UtcOffset::zero()),
+                        },
+                    ),
+                ),
+                ..*self
+            })
+        }
+
+        /// Sets the timestamp at which to interpret the time zone
+        /// for display name lookup.
```

Comment:
```suggestion
        /// Sets the timestamp, in milliseconds since Unix epoch, at which to interpret the time zone
        /// for display name lookup.
```